docs/build/advanced-concepts/randomness.md

```diff
@@ -176,49 +164,45 @@ access(all) fun revealCoinToss(receipt: @Receipt): @FungibleToken.Vault {
 
-While both are backed by Flow's Randomness Beacon it is important for developers to mindfully choose between `revertibleRandom` or
-seeding their own PRNG utilizing the `RandomBeaconHistory` smart contract:
+While both methods are backed by Flow's Randomness Beacon,
+it is important for developers to mindfully choose between `revertibleRandom` or seeding a PRNG using the `RandomBeaconHistory` smart contract:
 
-- With `revertibleRandom` a developer is calling the transaction environment,
-  which has the power to abort and revert if it doesn't like `revertibleRandom`'s outputs.
-  `revertibleRandom` is only suitable for smart contract functions that exclusively run within the trusted transactions.
-- In contrast, the `RandomBeaconHistory` contract is key for effectively implementing a commit-reveal scheme, where the transaction is non-trusted and may revert the random outputs.
+- With `revertibleRandom` a user has the power to abort and revert if it doesn't like `revertibleRandom`'s outputs.
+  `revertibleRandom` is only suitable for smart contract functions that exclusively run within trusted transactions emitted by trusted parties. You can think of a lottery contract picking a winning user, where the picking transaction is emitted by the lottery developer who is trusted not add the abortion logic into the transaction. Users are able to check the transaction code after it is submitted and make sure the lottery developer acted fairly.
+- In contrast, the commit-reveal method using the `RandomBeaconHistory` is necessary in cases where the transaction is submitted by non-trusted users and may revert the random outputs. 
+  You can think of a user minting a randomized NFT and is able to add a logic to their transaction to check the random traits and abandon the NFT if they are not happy. Another user plays a betting game and is able to abort whenever they lose. 
+  General users are not guaranteed to act honestly when they play and commit-reveal is the way to limit their actions. 
   During the commit phase, the user commits to proceed with a future source of randomness,
   which is only revealed after the commit transaction concluded. 
-  For each block, the `RandomBeaconHistory` automatically stores the generated source of randomness.
-  At the time of revealing the source, the committed source becomes a past-block source that can be queried through the history contract. 
 
 Adding a safe pattern to reveal randomness without the possibility of conditional transaction reversion unlocks applications relying on randomness. By providing examples of commit-reveal implementations we hope to foster a more secure ecosystem of decentralized applications and encourage developers to build with best practices.
 
-In simpler terms, the native secure randomness provided by the protocol can now be safely utilized within Cadence smart contracts
-and is available to all developers on Flow and the FVM.
-
 ## An Invitation to Build
```

Comment:
Not sure if we should keep this section, seems to be relevant during launch but not afterwards